core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -141,7 +141,7 @@ private[spark] class HighlyCompressedMapStatus private (
     private[this] var numNonEmptyBlocks: Int,
     private[this] var emptyBlocks: RoaringBitmap,
     private[this] var avgSize: Long,
-    @transient private var hugeBlockSizes: Map[Int, Byte])
+    private[this] var hugeBlockSizes: Map[Int, Byte])
```

Comment:
we do want to serialize `hugeBlockSizes`, but with customized logic, that why we marked it `@transient`.

I think the corrected fix is, make this class implements `KryoSerializable`, and copy the customized serialization logic for `hugeBlockSizes` to kryo serialization hooks.